borsh.go

```diff
@@ -379,18 +379,18 @@ func serialize(v reflect.Value, b io.Writer) error {
 	var err error
 	switch v.Kind() {
 	case reflect.Int8:
-		_, err = b.Write([]byte{byte((v.Interface().(int8)))})
+		_, err = b.Write([]byte{byte((v.Int()))})
 	case reflect.Int16:
 		tmp := make([]byte, 2)
-		binary.LittleEndian.PutUint16(tmp, uint16(v.Interface().(int16)))
+		binary.LittleEndian.PutUint16(tmp, uint16(v.Int()))
 		_, err = b.Write(tmp)
 	case reflect.Int32:
 		tmp := make([]byte, 4)
-		binary.LittleEndian.PutUint32(tmp, uint32(v.Interface().(int32)))
+		binary.LittleEndian.PutUint32(tmp, uint32(v.Int()))
 		_, err = b.Write(tmp)
 	case reflect.Int64:
 		tmp := make([]byte, 8)
-		binary.LittleEndian.PutUint64(tmp, uint64(v.Interface().(int64)))
+		binary.LittleEndian.PutUint64(tmp, uint64(v.Int()))
```

Comment:
The code here will still cause overflow.

Since the size of `int` is platform-dependent (can be 8-bit, 16-bit, 32-bit etc.), I don't think we should use `v.Int()` or `v.UInt()` to get value.  I recommend using `v.Convert()` to do the conversion.

Reply:
The `v.Int()` function will return a value which type is `int64`
It will fetch a proper range of memory and parse it into the biggest container

```
func (v Value) Int() int64 {
	k := v.kind()
	p := v.ptr
	switch k {
	case Int:
		return int64(*(*int)(p))
	case Int8:
		return int64(*(*int8)(p))
	case Int16:
		return int64(*(*int16)(p))
	case Int32:
		return int64(*(*int32)(p))
	case Int64:
		return *(*int64)(p)
	}
	panic(&ValueError{"reflect.Value.Int", v.kind()})
}
```

In this case, it always return the last condition `*(*int64)(p)`
I don't think it will cause overflow.
Correct me if I'm wrong.

Reply:
You are right, I thought the return type of `v.Int()` is `int`, turns out I was wrong.